File: Backend/maintenace-service/src/Entity/AreaFuncional.cs
namespace Entity
{
    public class AreaFuncional
    {
        public string Id { get; set; }              // Identificador único del área funcional
        public string IdPlanta { get; set; }        // FK de la planta asociada
        public string Nombre { get; set; }          // Nombre del área funcional
        public bool Estado { get; set; }            // Estado (activo/inactivo)
        public DateTime Fecha_log { get; set; }     // Fecha de creación o última modificación
    }
}

File: Frontend/src/Types/areafuncional.ts
export interface AreaFuncional {
  id: string;
  idPlanta: string;
  nombre: string;
  estado: boolean;
  fecha_log: string;
}

export interface AreaFuncionalResponse {
  data: AreaFuncional[];
  message: string;
  success: boolean;
}

export interface AreaFuncionalCreateRequest {
  idPlanta: string;
  nombre: string;
  estado: boolean;
  fecha_log: string;
}

export interface AreaFuncionalUpdateRequest extends AreaFuncionalCreateRequest {
  id: string;
}
